### PR TITLE
Authority-Select: Added a selection screen for choosing authority

### PR DESCRIPTION
**Pull request by kalpaitch**
Required for enforcement flow as in this PR and for partnership flow when applying for a new partnership.

### CODE REVIEW

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    $authority_options = [];
+    foreach($authorities as $authority) {
+      $authority_view = $authority_view_builder->view($authority, 'summary');
+
+      $authority_options[$authority->id()] = $this->renderMarkupField($authority_view)['#markup'];
+    }
+
+    // If a value already exists we can skip to the next step.
+    if ($this->getDefaultValues("par_data_authority_id", NULL)) {
+      return $this->redirect($this->getFlow()->getNextRoute('next'), $this->getRouteParams());
```
Comment (jon-edery) [inline web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php]: Should this if statement be executed before the foreach so it doesn't run if not needed?

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    $authorities= [];
+    if ($this->currentUser()->isAuthenticated()) {
+      $account = User::Load($this->currentUser()->id());
+      $authorities = $this->getParDataManager()->hasMembershipsByType($account, 'par_data_authority', TRUE);
+    }
+
+    $authority_view_builder = $this->getParDataManager()->getViewBuilder('par_data_authority');
+
+    $authority_options = [];
+    foreach($authorities as $authority) {
```
Comment (jon-edery) [inline web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php]: Space.  ^

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+   * Helper to get all the editable values when editing or
+   * revisiting a previously edited page.
+   *
+   * @param \Drupal\par_data\Entity\ParDataPartnership $par_data_partnership
+   *   The Partnership being retrieved.
+   */
+  public function retrieveEditableValues(ParDataPartnership $par_data_partnership = NULL) {
+
```
Comment (jon-edery) [inline web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php]: The implementation is missing or not needed ?

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, ParDataPartnership $par_data_partnership = NULL) {
+    $this->retrieveEditableValues();
```
Comment (jon-edery) [inline web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php]: Called for function above guessing this is not needed

File: web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php (hunk 3 above)
Comment (kalpaitch) [reply web/modules/features/par_partnership_flows/src/Form/ParPartnershipFlowsAuthoritySuggestionForm.php]: Not needed, the defaults are ok, but left in for consistency with other enforcement forms.

**Pull request merged**